core/src/main/java/com/linecorp/armeria/server/DefaultServiceConfigSetters.java

```diff
@@ -278,6 +281,13 @@ public ServiceConfigSetters errorHandler(ServiceErrorHandler serviceErrorHandler
         return this;
     }
 
+    @Override
+    public ServiceConfigSetters contextHook(Supplier<? extends AutoCloseable> contextHook) {
+        requireNonNull(contextHook, "contextHook");
+        this.contextHook = contextHook;
```

Comment:
Should we make this API take multiple contextHook like `NonWrappingRequestContext.hook()`?